src/SkynetServer/Network/DatabaseExtensions.cs

```diff
@@ -113,6 +113,51 @@ public static Task SendTo(this Packet packet, IEnumerable<long> accounts, Client
                 .Select(c => c.SendPacket(packet)));
         }
 
+        public static Task SendOrNotify(this Packet packet, IEnumerable<Session> sessions, Client exclude, long excludeFcm)
+        {
+            return Task.WhenAll(sessions.Select(async session =>
+            {
+                bool found = false;
+                foreach (Client client in Program.Clients)
+                {
+                    if (client.Session != null
+                        && client.Session.AccountId == session.AccountId
+                        && client.Session.SessionId == session.SessionId)
+                    {
+                        found = true;
+                        if (!ReferenceEquals(client, exclude))
+                            await client.SendPacket(packet);
+                        break;
+                    }
+                }
+                if (!found)
+                {
+                    if (session.AccountId != excludeFcm
+                        && !string.IsNullOrWhiteSpace(session.FcmToken)
+                        && session.LastFcmMessage < session.LastConnected)
+                    {
+                        try
+                        {
+                            await new FcmClient().SendAsync(session.FcmToken);
```

Comment:
Do we have to create a new `FcmClient` instance every single time we send a message? This does not seem to be very efficient as the `FcmClient` constructor loads FCM settings from a file every time.

Reply:
As `FcmClient.SendAsync` is not thread safe, we would have to use a client worker pool to achieve better performance. This change would be done in a separate pull request with a new Dependency Injection based architecture.

Reply:
Then maybe we can cache the settings within the `FcmClient` so that we do not have to load them from file every time?